core/src/saros/session/internal/SarosSession.java

```diff
@@ -724,6 +734,12 @@ public IReferencePoint getReferencePoint(String referencePointID) {
   @Override
   public void addReferencePointMapping(String referencePointId, IReferencePoint referencePoint) {
     if (referencePointMapper.getReferencePoint(referencePointId) == null) {
+      if (wouldCreateNestedReferencePoint(referencePoint)) {
+        leaveSessionDueToNestedReferencePoint(referencePoint);
```

Comment:
Just thrown an Exception.

Reply:
See other comment. In what state does this leave the session and the running resource negotiation?

Reply:
In the same broken state if you just press cancel during the resource negotiation. Still not fixed since years. Maybe this is a good start before thinking about other issues like this cornercase.

Reply:
I still suggest to move the introduced logic here to the ReferencePointMapper. And as I already stated just throw an exception for now. We already have an issue open for normal/abnormal cancellation of a current project negotiation. If you still want to abort the session do it in the IPN. But be aware that the IPN is executed on the server side in the Saros server project.

Reply:
Moving this check to the `ReferencePointMapper` is kind of messy. The check requires the `IReferencePointComparator` instance retrieved from the session context, but the mapper is instantiated before the context.

So I had to move the instantiation of the reference point mapper below the session context initialization.

Reply:
Another solution would be to move the method into the IReferencePoint interface.

Reply:
Yeah, that is a much better/cleaner approach. Will do. :+1:

Reply:
Moved the check to `SharedReferencePointMapper`. Also removes the logic leaving the session and just throws an exception.